### PR TITLE
gettext: use canonical relative paths for .pot

### PR DESCRIPTION
**Pull request by techtonik**
This removes unnecessary changes when rebuilding .pot files on Windows

### CODE REVIEW

File: sphinx/builders/gettext.py [hunk 1]
```diff
@@ -216,7 +216,8 @@ def finish(self):
                     if self.config.gettext_location:
                         # generate "#: file1:line1\n#: file2:line2 ..."
                         pofile.write("#: %s\n" % "\n#: ".join(
-                            "%s:%s" % (safe_relpath(source, self.outdir), line)
+                            "%s:%s" % (safe_relpath(source, self.outdir).
+                                       replace('\\', '/'), line)
```
Comment (lehmannro) [inline sphinx/builders/gettext.py]: Could you use `os.sep` and `sphinx.util.SEP` for this?  (Maybe, adding an inverse function for `sphinx.util.osutil.os_path` is appropriate?)

**Comment by shimizukawa** [comment]
@techtonik Looks Good. Can you please squash this pull request before merging?

**Comment by techtonik** [comment]
Squashed flakery.

**Comment by shimizukawa** [comment]
@techtonik Thanks!

**Pull request merged**